src/main/java/org/truffleruby/interop/RubyMessageResolution.java

```diff
@@ -71,6 +71,9 @@ private RubyContext getContext() {
         @Child private DoesRespondDispatchHeadNode doesRespond = DoesRespondDispatchHeadNode.create();
 
         protected Object access(VirtualFrame frame, DynamicObject object) {
+            if (RubyGuards.isRubyString(object)) {
+                return false;
+            }
```

Comment:
This was a workaround to get around the stack overflow issue, but I believe we should have HAS_SIZE==true only for Array. Can be in another PR, as you prefer.

Reply:
I'll try removing it for hashes as well.

Reply:
I think it should be just `return RubyGuards.isRubyArray(object)`.
And we can extend in the future if needed if there is another array-like kind of object.